vesting/lib.rs

```diff
@@ -47,14 +60,16 @@ mod vesting_contract {
         pub fn new(
             beneficiary: AccountId,
             duration_time_in_sec: Timestamp,
-        ) -> Result<Self, Error> {
+        ) -> Result<Self> {
             if beneficiary == AccountId::from([0x0; 32]) {
                 return Err(Error::InvalidBeneficiary);
             }
             
-            // This is multiplied by 1000 to conform to the
+            // This is multiplied by 1000 to conform to the 
             // Timestamp fomat in ink.
-            let duration_time = duration_time_in_sec * 1000;
+            let duration_time = duration_time_in_sec
+                .checked_mul(1000)
+                .ok_or(Error::Overflow)?;
```

Comment:
This could be avoided by making `duration_time_in_sec` a `u32` which still allows durations > 100 years.

Or simply explicitly check that `duration_time_in_sec` < u64::MAX / 1000 and throw a more specific error e.g. `Error::InvalidDuration`.

Reply:
> This could be avoided by making `duration_time_in_sec` a `u32` which still allows durations > 100 years.
> 
> Or simply explicitly check that `duration_time_in_sec` < u64::MAX / 1000 and throw a more specific error e.g. `Error::InvalidDuration`.

done